src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -161,10 +164,43 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
       }
     };
 
-    cy.on('tap', (evt: any) => {
-      const cytoscapeEvent = getCytoscapeBaseEvent(evt);
-      if (cytoscapeEvent) {
-        this.handleTap(cytoscapeEvent);
+    const injectDoubleTap = fun => {
+      let lastTap, currentTap, tapTimeout;
+
+      return (evt: any) => {
+        currentTap = evt.target;
+
+        tapTimeout = setTimeout(() => (lastTap = undefined), 350);
```

Comment:
Also, I think this could trigger the "single" click too (maybe), not sure if it will behave OK
`setTimeout(() => {lastTap = undefined; fun(evt)}, 350);`
So a double click won't trigger a single click (which is how i think it should behave)
The problem is that the delay (350ms) might be noticeable and annoying.

Reply:
What do you mean by "reduce the scope"? I agree with you that the double click should not trigger the click event twice, will change that.

As for the delay, I agree, maybe 100ms would be better?

Reply:
Sorry, i meant to say that we should declare `tapTimeout` inside of `return(evt:any) => {` instead of the outer block

Reply:
About declaring the tapTimeout inside the function: I checked the code, the problem doing it is that I need the reference to outlive the function, to catch the second click correctly.

And the delay problem, I think it will not happen, because it waits on its own thread, and if you click anywhere else it resets the function.

Reply:
> About declaring the tapTimeout inside the function: I checked the code, the problem doing it is that I need the reference to outlive the function, to catch the second click correctly.

right, missed that.

> And the delay problem, I think it will not happen, because it waits on its own thread, and if you click anywhere else it resets the function.

My concern is that the single click will lag for 350ms ( or 100ms if you change that), but i suppose we can test, thanks.